library/pod-security-policy/forbidden-sysctls/src.rego

```diff
@@ -0,0 +1,26 @@
+package k8spspforbiddensysctls
+
+violation[{"msg": msg, "details": {}}] {
+    sysctls := {x | x = input.review.object.spec.securityContext.sysctls[_][name]}
+    input_sysctls(sysctls)
+    msg := sprintf("One of the sysctls %v is not allowed, pod: %v. Forbidden sysctls: %v", [sysctls, input.review.object.metadata.name, input.parameters.forbiddenSysctls])
+}
+
+# * may be used to forbid all sysctls
+input_sysctls(sysctls) {
+    input.parameters.forbiddenSysctls[_] == "*"
+}
+
+input_sysctls(sysctls) {
+    forbidden_sysctls := {x | x = input.parameters.forbiddenSysctls[_]}
+    test := sysctls & forbidden_sysctls
+    count(test) > 0
+}
+
+input_sysctls(sysctls) {
+    prefix_matches(input.parameters.forbiddenSysctls[_], sysctls[_])
+}
+
+prefix_matches(prefix, sysctl) {
+    contains(sysctl, trim(prefix, "*"))
```

Comment:
What about the case where `*` is not provided? Will a prefix match still be performed?

Also, shouldn't this be `startswith` instead of `contains`?

Reply:
It needs `*` at the end if we want to match the PSP behavior. 

> A sysctl pattern ends with a * character, such as kernel.*. 
https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/#podsecuritypolicy

Reply:
Right, it looks like as-written a `*` is not required, as it trims `*` from the prefix without checking for its existence, meaning that both inputs containing `*` and those not containing `*` will be prefix checked.

Also, `startswith` is more proper as contains would find prefix "hello" in "okayhello", even though it's a suffix.

Reply:
thanks! updated.